lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4274,7 +4274,8 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
       if (sourceLoc.isValid()) {
         getContext()
             .diagnose(sourceLoc, diag::autodiff_nonvaried_result_fixit)
-            .fixItInsertAfter(sourceLoc, ".withoutDerivative()");
+            .fixItInsert(sourceLoc, "withoutDerivative(at:")
+            .fixItInsertAfter(sourceLoc, ")");
```

Comment:
You might need to lookup an expression first and get its source range, from which you can get its start loc and end loc.

Reply:
I just pushed a fix. Would that work?

Reply:
Looks like it might work. There may be many tricky edge cases. Did you update tests?

Reply:
Yes I can see that. I hope that by working on incremental fixes and features in the compiler codebase I may gain a basic understanding of what's going on overall and what the common/useful patterns are.